controller/backup_target_controller.go

```diff
@@ -877,12 +877,16 @@ func (btc *BackupTargetController) isResponsibleFor(bt *longhorn.BackupTarget, d
 		return false, err
 	}
 
-	instanceManager, err := btc.ds.GetRunningInstanceManagerByNodeRO(btc.controllerID, "")
-	if err != nil {
-		return false, err
-	}
-	if instanceManager == nil {
-		return false, errors.New("failed to get running instance manager")
+	// Avoid the situation where the instance manager pods are already deleted but the BackupTarget is still there, and try to delete the BackupTarget.
+	// For example, uninstall Longhorn after all data engines are disabled.
+	if bt.DeletionTimestamp.IsZero() {
```

Comment:
This change means that we will skip checking instance managers only when the backup target CR is marked as deleted.

My question is, why should we continue returning `errors.New("failed to get running instance manager")` when no running instance manager is found?

Reply:
>continue returning errors.New("failed to get running instance manager")

I think you meant line 888 when `instanceManager == nil`
From line 882 to line 890, it checks if the instance manager is running for the backup target that is not marked deleted, and it's from https://github.com/longhorn/longhorn-manager/pull/2764.
We might not have the possibility to reconcile the backup target again without returning an error after the instance manager is running.
@derekbit can you explain more about this PR?

Reply:
The original implementation will be stuck like the uninstallation issue ticket (longhorn/longhorn#11934)

If BT deletion is triggered, the IM state check will be skipped in this PR, and then the backup target controller (the backup target owner) will start the deletion process in the reconciliation.